src/collectionData.js

```diff
@@ -0,0 +1,109 @@
+      listDescription.type = 'text';
+      listDescription.className = 'todo-list-item-description';
+      listDescription.value = task.description;
+      const removeBtn = document.createElement('button');
+      removeBtn.className = 'todo-list-item-remove';
+      removeBtn.dataset.id = task.taskIndex;
+      const newListTag = document.createElement('li');
+      newListTag.className = 'todo-list-item';
+      newListTag.id = `list-item${task.taskIndex}`;
+
+      // implement drag and drop functionality
+      newListTag.draggable = true;
+      newListTag.addEventListener('dragstart', drag);
+      newListTag.addEventListener('drop', drop);
+      newListTag.addEventListener('dragover', allowDrop);
+      newListTag.append(checkbox, listDescription, removeBtn);
+      todoListWrapperElement.appendChild(newListTag);
+      this.taskRemoveButtons.push(removeBtn);
+    });
+
+    // toggle isCompleted to true/false when checkbox is checked/unchecked
+    const todoListWrapperChildren = Array.from(todoListWrapperElement.children);
+    todoListWrapperChildren.forEach((child, index) => {
+      child.children[0].addEventListener('click', () => {
+        if (child.children[0].checked) {
+          child.children[1].style.textDecoration = 'line-through';
+          this.todoListTasks[index].isCompleted = true;
+        } else {
+          child.children[1].style.textDecoration = 'none';
+          this.todoListTasks[index].isCompleted = false;
+        }
+      });
+    });
+
+    // add event listener to input field to toggle styling and update task description
+    const listDescriptionArr = document.getElementsByClassName('todo-list-item-description');
+    const inputFields = Array.from(listDescriptionArr);
+    inputFields.forEach((input, i) => {
+      input.addEventListener('click', () => {
+        if (document.activeElement === input) {
+          todoListWrapperElement.children[i].style.backgroundColor = 'yellow';
+          todoListWrapperElement.children[i].children[2].style.display = 'block';
+          input.style.backgroundColor = 'yellow';
+          todoListWrapperChildren[i].style.setProperty('--before', 'none');
+        } else {
+          todoListWrapperElement.children[i].style.backgroundColor = 'white';
+          todoListWrapperElement.children[i].children[2].style.display = 'none';
+          input.style.backgroundColor = 'white';
+          todoListWrapperChildren[i].style.setProperty('--before', 'block');
+        }
+      });
+
+      input.addEventListener('change', () => {
+        this.todoListTasks[i].description = input.value;
+        todoListWrapperElement.children[i].style.backgroundColor = 'white';
+        todoListWrapperElement.children[i].children[2].style.display = 'none';
+        input.style.backgroundColor = 'white';
+        todoListWrapperChildren[i].style.setProperty('--before', 'block');
+        document.getElementById('todo-input').focus();
+      });
+    });
```

Comment:
- [ ] Kindly make sure after editing the task description, you update the local storage. At this moment the user can be able to edit but the changes are not being saved in the local storage. Kindly fix that.

Reply:
fixed